Cgen error when using "$fn"

### Describe the bug

As far as I know, "$fn" isn't a valid. But when I used it, it tried to compile anyway and made a cgen error.

### Expected Behavior

For it to print a compiler message saying $fn is invalid

### Current Behavior

/tmp/v_1000/main.15353534442425981228.tmp.c:12442: error: #endif without matching #if

### Reproduction Steps

```v
module main

$fn func(a int) int {
    return a + a
}

fn main() {
    println(func(5))
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V full version: V 0.3.3 2179db3.a938dcd

### Environment details (OS name and version, etc.)

OS: linux, Linux Mint 20.3
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz

--- a/vlib/v/checker/tests/invalid_comptime_test.out
+++ b/vlib/v/checker/tests/invalid_comptime_test.out
@@ -1,0 +1,5 @@
+vlib/v/checker/tests/invalid_comptime_test.vv:1:1: error: unexpected token `$`
+    1 | $fn func(a int) int {
+      | ^
+    2 |     return a + a
+    3 | }

--- a/vlib/v/checker/tests/invalid_comptime_test.vv
+++ b/vlib/v/checker/tests/invalid_comptime_test.vv
@@ -1,0 +1,7 @@
+$fn func(a int) int {
+    return a + a
+}
+
+fn main() {
+    println(func(5))
+}

--- a/vlib/v/parser/parser.v
+++ b/vlib/v/parser/parser.v
@@ -754,7 +754,7 @@ pub fn (mut p Parser) top_stmt() ast.Stmt {
 				if p.peek_tok.kind == .key_for {
 					comptime_for_stmt := p.comptime_for()
 					return p.other_stmts(comptime_for_stmt)
-				} else {
+				} else if p.peek_tok.kind == .key_if {
 					if_expr := p.if_expr(true)
 					cur_stmt := ast.ExprStmt{
 						expr: if_expr
@@ -765,6 +765,8 @@ pub fn (mut p Parser) top_stmt() ast.Stmt {
 					} else {
 						return p.other_stmts(cur_stmt)
 					}
+				} else {
+					return p.unexpected()
 				}
 			}
 			.hash {

